crates/gpui/src/platform/windows/platform.rs

```diff
@@ -290,17 +298,91 @@ impl Platform for WindowsPlatform {
 
     // todo!("windows")
     fn open_url(&self, url: &str) {
-        // todo!("windows")
+        unimplemented!()
     }
 
     // todo!("windows")
     fn on_open_urls(&self, callback: Box<dyn FnMut(Vec<String>)>) {
         self.inner.callbacks.lock().open_urls = Some(callback);
     }
 
-    // todo!("windows")
+    #[allow(overflowing_literals)]
     fn prompt_for_paths(&self, options: PathPromptOptions) -> Receiver<Option<Vec<PathBuf>>> {
-        unimplemented!()
+        println!("prompt_for_paths {:?}", options);
+
+        let (tx, rx) = oneshot::channel();
+
+        self.foreground_executor().spawn(async move {
+            // initialize COM
+            let hr = unsafe { CoInitializeEx(Some(std::ptr::null_mut()), COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE) };
+            if hr.is_err() { panic!("CoInitializeEx failed: {:?}", hr); }
+
+            let tx = Cell::new(Some(tx));
+
+            // create file open dialog
+            let folder_dialog: IFileOpenDialog = unsafe {
+                CoCreateInstance::<std::option::Option<&IUnknown>, IFileOpenDialog>(
+                    &FileOpenDialog,
+                    None,
+                    CLSCTX_ALL
+                ).unwrap()
+            };
+
+            // dialog options (https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/ne-shobjidl_core-_fileopendialogoptions)
+            let mut dialog_options: u32 = 0x1000; // FOS_FILEMUSTEXIST
+
+            if options.multiple {
+                dialog_options |= 0x200; // FOS_ALLOWMULTISELECT
+            }
+
+            if options.directories {
+                dialog_options |= 0x20; // FOS_PICKFOLDERS
+            }
+
+            unsafe {
+                folder_dialog.SetOptions(FILEOPENDIALOGOPTIONS(dialog_options)).unwrap(); // FOS_PICKFOLDERS
+                folder_dialog.SetTitle(&HSTRING::from(OsString::from("Select a folder"))).unwrap();
+            }
+
+            let hr = unsafe { folder_dialog.Show(None) };
+
+            if hr.is_err() {
+                if hr.unwrap_err().code() == HRESULT(0x800704C7) { // user canceled error (we don't want to crash)
+                    if let Some(tx) = tx.take() {
+                        tx.send(None).unwrap();
+                    }
+                    return;
+                }
+            }
+
+            let mut results = unsafe { folder_dialog.GetResults().unwrap() };
+
+            let mut paths: Vec<PathBuf> = Vec::new();
+            for i in 0..unsafe { results.GetCount().unwrap() } {
+                let mut item: IShellItem = unsafe { results.GetItemAt(i).unwrap() };
+                let mut path = unsafe { item.GetDisplayName(SIGDN(0x80058000)).unwrap() }; // SIGDN_FILESYSPATH
+
+                paths.push(PathBuf::from(unsafe { path.to_string().unwrap() }));
```

Comment:
I don't think this is safe on windows? IIRC windows UTF-16 and rust strings are UTF-8. At least, this should be `to_string_lossy()`